File: app/lib/features/relation/domain/relation.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'relation.freezed.dart';
part 'relation.g.dart';

/// Relation 实体 — "织线"
@freezed
class Relation with _$Relation {
  const factory Relation({
    required int id,
    @Default('') String publicId,
    @Default(0) int userId,
    required int sourceFragmentId,
    required int targetFragmentId,
    @Default('reminds_me') String relationType,
    String? note,
  }) = _Relation;

  factory Relation.fromJson(Map<String, dynamic> json) =>
      _$RelationFromJson(json);
}

File: backend/internal/relation/domain/relation.go
package domain

import "time"

type Relation struct {
	ID               int64     `json:"id"`
	PublicID         string    `json:"public_id"`
	UserID           int64     `json:"user_id"`
	SourceFragmentID int64     `json:"source_fragment_id"`
	TargetFragmentID int64     `json:"target_fragment_id"`
	RelationType     string    `json:"relation_type"`
	Note             string    `json:"note"`
	CreatedAt        time.Time `json:"created_at"`
}

type CreateParams struct {
	SourceFragmentID int64
	TargetFragmentID int64
	RelationType     string
	Note             string
}
